### DISCUSSION TITLE
The vanilla store selectors in the zustand documentation are not available without react.

### DISCUSSION THREAD

**Discussion by R0boter** [pmndrs/zustand#2752] [Q&A] [answered]
I did't know this is document error, or it's an issue,
I want use zustand without react, so i use the code with [document](https://zustand.docs.pmnd.rs/guides/auto-generating-selectors#vanilla-store)
but it's also report Error
Error
Invalid hook call. Hooks can only be called inside of the body of a function component.
this is the [minimum Reproduction File](https://codesandbox.io/p/sandbox/cool-pascal-c99g5r)
the triggers error's code in the http.js

if it's document error, how can i use zustand selector without react?

**Comment by dai-shi** [comment]
> I want use zustand without react

You need to import from `zustand/vanilla`.

https://github.com/pmndrs/zustand/blob/f82f56ddebf766e16e60156a28e216c0d1eec58c/readme.md?plain=1#L224-L235

**Reply by R0boter** [reply]
yes, i used,there is i created [minimum reproduction project](https://codesandbox.io/p/sandbox/cool-pascal-c99g5r?file=%2Fsrc%2Fauth.js%3A16%2C1)

**Reply by R0boter** [reply]
> > I want use zustand without react
> 
> You need to import from `zustand/vanilla`.
> 
> https://github.com/pmndrs/zustand/blob/f82f56ddebf766e16e60156a28e216c0d1eec58c/readme.md?plain=1#L224-L235

i readed the document over again, it means create selectors need use `useStore` this hook, if i want use store without react, i can't use the selector generated by useStore, i just can use store.getState().xxx without react, when i in react components i can use the selector generated by the useStore
is that what the document means?
thank you for your answers

**Reply by dai-shi** [reply, marked as answer]
If you don't use React, you don't need useStore hook. You can just select the store state.
```js
const selector = (state) => state.xxx;
const slice = selector(store.getState());
```

**Reply by R0boter** [reply]
i think i already know the answer, thanks for the answer

**Comment by dbritto-dev** [comment]
@R0boter here you go -> https://zustand.docs.pmnd.rs/apis/create-store

**Reply by R0boter** [reply]
yes, i used,there is i created [minimum reproduction project](https://codesandbox.io/p/sandbox/cool-pascal-c99g5r?file=%2Fsrc%2Fauth.js%3A16%2C1)

**Reply by R0boter** [reply]
> @R0boter here you go -> https://zustand.docs.pmnd.rs/apis/create-store

i readed the document over again, it means create selectors need use `useStore` this hook, if i want use store without react, i can't use the selector generated by useStore, i just can use store.getState().xxx without react, when i in react components i can use the selector generated by the useStore
is that what the document means?
thank you for your answers

**Reply by dbritto-dev** [reply]
@R0boter you got it

**Reply by R0boter** [reply]
> @R0boter you got it

thanks you for your answer
